### PR TITLE
json: fix a bug in path_expr.

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

**Comment by hicqu** [comment]
@hhkbp2 @tiancaiamao , PTAL, thanks.

File: util/types/json/path_expr.go [hunk 1]
```diff
@@ -81,8 +81,7 @@ const (
 
 // containsAnyAsterisk returns true if pef contains any asterisk.
 func (pef pathExpressionFlag) containsAnyAsterisk() bool {
-	pef &= pathExpressionContainsAsterisk
```
Comment (XuHuaiyu) [inline util/types/json/path_expr.go]: will change `&=` to use `|=` be better? @hicqu

Comment (hicqu) [inline util/types/json/path_expr.go]: If so, how about pef contains other flags? `|=` can't filter those.

Comment (XuHuaiyu) [inline util/types/json/path_expr.go]: ok, I misunderstood this function.

**Comment by hanfei1991** [comment]
add a test that return error of "Invalid path expression" instead of modifying successfully.

**Comment by hhkbp2** [comment]
LGTM

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**